docs/common/project/how-to-release.md

```diff
@@ -255,3 +260,63 @@ Java artifacts are available from Maven Central.
 
 Thanks to everyone for contributing!
 ```
+
+### Documentation Release
+
+Documentation needs to be updated as a part of Iceberg release after a release candidate is passed.
+The commands described below assume the `iceberg-docs` repository and `iceberg` repository are in the same parent directory locally, 
+and the release manager is executing commands in the `iceberg` repository.
+Adjust the commands accordingly if it is not the case.
+
+#### Release notes
+
+Release notes need to be drafted and published in the `iceberg` repository and reviewed by the community.
+
+#### Iceberg version update
+
+Before release, publish a PR in the `iceberg-docs` repository to update `landing-page/config.toml` and `docs/config.toml` to use the new Iceberg version number.
+
+#### Documentation update
+
+To start the release process, run the following steps in the `iceberg-docs` repository to copy docs over:
+
+```shell
+rm -rf ../iceberg-docs/docs/content/docs
+rm -rf ../iceberg-docs/landing-page/content/common
+cp -r docs/versioned ../iceberg-docs/docs/content/docs
+cp -r docs/common ../iceberg-docs/landing-page/content/common
+```
+
+The resulted changes in `iceberg-docs` should be approved in a separated PR.
+
+#### Javadoc update
+
+In the `iceberg` repository, generate the javadoc for your release and copy it to the `javadoc` folder in `iceberg-docs` repo:
+```shell
+./gradlew refreshJavadoc
+rm -rf ../iceberg-docs/javadoc
+cp site/docs/javadoc/<VERSION NUMBER> ../iceberg-docs/javadoc
+```
+
+This resulted changes in `iceberg-docs` should be approved in a separated PR.
+
+#### Cut a new version branch
+
+Once completed, go to the `iceberg-docs` repository (the Apache one, not a forked one) to cut a new branch using the version number as the branch name.
+For example, to cut a new versioned doc for release `0.13.0`:
+
+```shell
+git checkout -b 0.13.0
```

Comment:
@samredai, I thought just the 0.13.0 docs were in the branch. Is the branch a complete copy of everything in master?

Reply:
Yes the branch is a full branch from master/main and points to the commit at the time of release. That triggers the deployment workflow which builds the site and deploys it to the asf-site branch (the built site only) under the docs/<version> directory. The unbuilt site has to be maintained to allow for hotfixes in the future.

Reply:
currently it is